### DISCUSSION TITLE
Hybrid Filter-Action trigger type

### DISCUSSION THREAD

**Discussion by gdpdils** [directus/directus#17462] [Feature Requests]
### Summary

At the moment we can define flow triggers either as _filters_ or as _actions_:

|                      | Filter          | Action       |
| -------------------- | ------------  | ------------ |
| Blocking             |       X       |              |
| After the operation |               |     X        |

While we don't really see a use case for _non-blocking_ triggers that execute _before_ the triggering operation, there's clearing a spot for _blocking_ actions to be executed _after_ the operation they're attached to:

|                      | Filter          | Action       | ❓ |
| -------------------- | ------------  | ------------ | --- | 
| Blocking             |       X       |              | X |
| After the operation |               |     X        | X  | 

### Basic Example

`Flow A` triggers whenever a new item is _added_ to `Collection A` and `Flow B` triggers whenever any item is _updated_ on `Collection B`.

`Flow A` calls a custom endpoint that executes some code and ultimately updates some items on `Collection B`. This, in turn, triggers `Flow B` (that, for instance, deletes some other items in `Collection C` and sends an email)

We need a new type of trigger then as the result of `POST`ing a new item to `Collection A` returns only when the whole chain of events "Inserting the new item in Collection A" -> "Executing Flow A" -> "Executing Flow B" has been atomically executed so that receiving an OK response to the "insert into Collection A" call means not only that the new item is now in `Collection A`, but also that some items are actually updated on `Collection B`, some are deleted in `Collection C` and an email has been sent. All successfully.

### Motivation

In our real world experience of extensively using Directus flows, we've found out that some 90% of the times we are just hacking our way around the current implementation because what we actually need is a third kind of trigger that _synchronously_ runs _after_ the operation it's triggered by has finished executing.

For instance, we want that whenever a new item has been added to a certain collection (meaning that the insert operation is successful and we have access to the id of the new item in the code), some operation is ran, typically the execution of custom code. While this sounds like an _action_ trigger, we want it to be _blocking_.

Now, with an _action_ filter, the result of creating a new item via an API call is that the call immediately returns after the insert is done. The client is left in the dark about whether the attached flow(s) will execute, when, when they are done running and if they'll be successful or not.

### Detailed Design

We guess this should use the "execute after" code from _action_ trigger and the "blocking" code from _filter_ triggers.

### Must Have:
When creating a flow, it should be possible to select a third kind of trigger that is "post-blocking"

### Drawbacks

No drawbacks, it's a new feature.

### Alternatives

This is actually an alternative to the two trigger types currently existing 

### Adoption Strategy

We're not familiar with the development cycle of Directus.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
